apps/src/lib/levelbuilder/SaveBar.jsx

```diff
@@ -1,71 +1,82 @@
+      <button
+        className="btn"
+        type="button"
+        style={isShowButtonRendered ? styles.saveButton : styles.hide}
+        onClick={handleView || (() => navigateToHref(pathForShowButton))}
+        disabled={isSaving}
+      >
+        Show
+      </button>
+      <div style={styles.saveButtonArea}>
+        {lastSaved && !error && (
+          <div style={styles.lastSaved} className="lastSavedMessage">
+            {`Last saved at: ${new Date(lastSaved).toLocaleString()}`}
+          </div>
+        )}
+        {error && <div style={styles.error}>{`Error Saving: ${error}`}</div>}
+        {isSaving && (
+          <div style={styles.spinner}>
+            <FontAwesome icon="spinner" className="fa-spin" />
+          </div>
+        )}
         <button
           className="btn"
           type="button"
           style={styles.saveButton}
-          onClick={this.props.handleView}
-          disabled={this.props.isSaving}
+          onClick={e => handleSave(e, false)}
+          disabled={isSaving}
         >
-          Show
+          Save and Keep Editing
+        </button>
+        <button
+          className="btn btn-primary"
+          type="submit"
+          style={styles.saveButton}
+          onClick={e => handleSave(e, true)}
+          disabled={isSaving}
+        >
+          Save and Close
         </button>
-        <div style={styles.saveButtonArea}>
-          {this.props.lastSaved && !this.props.error && (
-            <div style={styles.lastSaved} className="lastSavedMessage">
-              {`Last saved at: ${new Date(
-                this.props.lastSaved
-              ).toLocaleString()}`}
-            </div>
-          )}
-          {this.props.error && (
-            <div style={styles.error}>{`Error Saving: ${
-              this.props.error
-            }`}</div>
-          )}
-          {this.props.isSaving && (
-            <div style={styles.spinner}>
-              <FontAwesome icon="spinner" className="fa-spin" />
-            </div>
-          )}
-          <button
-            className="btn"
-            type="button"
-            style={styles.saveButton}
-            onClick={e => this.props.handleSave(e, false)}
-            disabled={this.props.isSaving}
-          >
-            Save and Keep Editing
-          </button>
-          <button
-            className="btn btn-primary"
-            type="submit"
-            style={styles.saveButton}
-            onClick={e => this.props.handleSave(e, true)}
-            disabled={this.props.isSaving}
-          >
-            Save and Close
-          </button>
-        </div>
       </div>
-    );
-  }
+    </div>
+  );
 }
 
+SaveBar.propTypes = {
+  error: PropTypes.string,
+  handleSave: PropTypes.func.isRequired,
+  handleView: PropTypes.func,
+  isSaving: PropTypes.bool,
+  lastSaved: PropTypes.number,
+  pathForShowButton: PropTypes.string
+};
+
+SaveBar.defaultProps = {
```

Comment:
Yeah, good questions. I suppose we should norm on best practices, and it sounds like there already is a norm for `isRequired`. I am accustomed to (a) only marking `isRequired` for props that folks using the component need to provide, and then (b) providing defaults for all not-required props so that people can easily tell what's happening by default. I suppose I don't feel strongly about (b), but feel more strongly about (a): it feels like marking `isRequired` for all of these, but then only having an error if a person doesn't provide handleSave, could be confusing. Then again, we have tests that pretty clearly show that, and it sounds like that's a different norm from what we're already doing

Reply:
nice, I like that better - it does seem more useful that the isRequired props would only be the ones the caller has to provide. let's go with that!

Reply:
Other thoughts about removing the `undefined` defaultProps? I'm good to do so––curious to hear what folks think.